AmplifyPlugins/API/Tests/APIHostApp/AWSAPIPluginFunctionalTests/GraphQLConnectionScenario1Tests.swift

```diff
@@ -190,49 +179,32 @@ class GraphQLConnectionScenario1Tests: XCTestCase {
         XCTAssertEqual(resultsArray.count, 2)
     }
 
-    func createTeam(id: String = UUID().uuidString, name: String) -> Team1? {
+    func createTeam(id: String = UUID().uuidString, name: String) async throws ->  Team1? {
         let team = Team1(id: id, name: name)
         var result: Team1?
-        let requestInvokedSuccessfully = expectation(description: "request completed")
-        Amplify.API.mutate(request: .create(team)) { event in
+        let event = try await Amplify.API.mutate(request: .create(team))
             switch event {
-            case .success(let data):
-                switch data {
                 case .success(let team):
                     result = team
-                default:
-                    XCTFail("Could not get data back")
-                }
-                requestInvokedSuccessfully.fulfill()
-            case .failure(let error):
-                XCTFail("Failed \(error)")
+            case .failure(let graphQLResponseError):
+                    XCTFail("Failed with error: \(graphQLResponseError)")
             }
-        }
-        wait(for: [requestInvokedSuccessfully], timeout: TestCommonConstants.networkTimeout)
         return result
     }
 
     func createProject(id: String = UUID().uuidString,
-                       name: String? = nil,
-                       team: Team1? = nil) -> Project1? {
+                        name: String? = nil,
+                        teamID: String,
+                        team: Team1? = nil) async throws -> Project1? {
         let project = Project1(id: id, name: name, team: team)
         var result: Project1?
-        let requestInvokedSuccessfully = expectation(description: "request completed")
-        Amplify.API.mutate(request: .create(project)) { event in
+        let event = try await Amplify.API.mutate(request: .create(project))
             switch event {
-            case .success(let data):
-                switch data {
-                case .success(let project):
+            case .success(let project):
                     result = project
-                default:
-                    XCTFail("Could not get data back")
+            case .failure(let graphQLResponseError):
+                    XCTFail("Failed with error: \(graphQLResponseError)")
```

Comment:
Failing here works as well and has some intention behaviors for our testing since a call to `createProject` which receives a GraphQLResponseError will then hit this line, `XCTFail`, to fail the test. Failing the test is not always the case since tests can be configured to continue on failure. I believe if you throw `graphQLResponseError` then the code will happily compile since all code paths will complete the method. Having this `XCTFail` is probably the reason why we are holding `var result: Project1?` and returning `nil`, which is required since XCTFail does not complete the method.

Reply:
```suggestion
   throw graphQLResponseError
```